File: 134. Gas Station/Solution.py
class Solution:
    # @param {integer[]} gas
    # @param {integer[]} cost
    # @return {integer}
    def canCompleteCircuit(self, gas, cost):
        if len(gas) == 0 or len(cost) == 0:
            return -1
            
        startPoint = 0
        curGas = 0
        totalGas = 0
        
        for i in xrange(len(gas)):
            curGas += gas[i] - cost[i]
            totalGas += gas[i] - cost[i]
            
            if curGas < 0:
                curGas = 0
                startPoint = i+1
                
        return startPoint if totalGas >= 0 else -1

File: 134. Gas Station/Solution.java
public class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        if (gas == null || cost == null || gas.length == 0 || cost.length == 0)
            return -1;
            
        int curGas = 0;
        int totalGas = 0;
        int startPoint = 0;
        
        for (int i = 0; i < gas.length; i++) {
            curGas += gas[i] - cost[i];
            totalGas += gas[i] - cost[i];
            
            if (curGas < 0) {
                curGas = 0;
                startPoint = i+1;
            }
        }
        
        return totalGas >= 0? startPoint: -1;
    }
}